yb-voyager/src/utils/sqlname/sqlname.go

```diff
@@ -311,3 +313,39 @@ func IsReservedKeywordPG(word string) bool {
 func IsReservedKeywordOracle(word string) bool {
 	return slices.Contains(OracleReservedKeywords, word)
 }
+
+type ObjectNameQualifiedWithTableName struct {
+	SchemaName        string
+	TableName         string
+	ObjectName        string
+	FromDefaultSchema bool
+
+	Qualified    identifier
+	Unqualified  identifier
+	MinQualified identifier
+}
+
+func NewObjectNameQualifiedWithTableName(dbType, defaultSchemaName, objectName string, schemaName, tableName string) *ObjectNameQualifiedWithTableName {
+	result := &ObjectNameQualifiedWithTableName{
+		SchemaName:        schemaName,
+		FromDefaultSchema: schemaName == defaultSchemaName,
+		TableName:         tableName,
+		ObjectName:        objectName,
+		Qualified: identifier{
+			Quoted:    quote2(dbType, objectName) + " ON " + schemaName + "." + quote2(dbType, tableName),
```

Comment:
shall we keep the identifiers properly formatted as  `<schema_name>.<table_name>.<column_name>` ?
If we want the string representation to be "column On schema.table" , we can define it as String()?